Added the vocation base id and the functions to manage it instead of the "client id"

beats-dh opened the pull request:
New lua function: vocation:getBaseId()
New c++ function vocation->getBaseId()

The idea of the base id is that the "promotions" have the same id as the base vocation, so that it is not necessary to have repetitions of ifs and if a new vocation is added, it will also not be necessary to register them in the scripts that call the vocations , because the id is the same. This function should only be used in scripts that will be used by all base vocations.

For example, if a spell can be buy by the "knight" or "elite knight", then you can use vocation:getBaseId()
It is always preferable to use vocation:getBaseId() when possible.

lgrossi commented on src/utils/enums.h:
```diff
@@ -821,6 +822,69 @@ struct CombatDamage
+	LUA_ERROR_SPELL_NOT_FOUND
+};
+
+struct LootBlock;
```
can you please move those structs to a new header file?

dudantas replied:
> can you please move those structs to a new header file?

![image](https://user-images.githubusercontent.com/8551443/116612446-ef562000-a90d-11eb-9715-009a87a8eb14.png)

beats-dh committed:
add BaseId

New lua function: vocation:getBaseId()
New c++ function vocation->getBaseId()

The idea of the base id is that the "promotions" have the same id as the base vocation, so that it is not necessary to have repetitions of ifs and if a new vocation is added, it will also not be necessary to register them in the scripts that call the vocations , because the id is the same. This function should only be used in scripts that will be used by all base vocations.

For example, if a spell can be buy by the "knight" or "elite knight", then you can use vocation:getBaseId()
It is always preferable to use vocation:getBaseId() when possible.

beats-dh committed:
Fix potions.lua
